### PR TITLE
ARTEMIS-2692 update dependency for JMS client

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

**Comment by jbertram** [comment]
@clebertsuconic, this should probably be merged before the release.

File: artemis-jms-client/pom.xml [hunk 1]
```diff
@@ -50,6 +50,11 @@
          <artifactId>artemis-core-client</artifactId>
          <version>${project.version}</version>
       </dependency>
+      <dependency>
+         <groupId>org.apache.activemq</groupId>
+         <artifactId>artemis-commons</artifactId>
```
Comment (michaelandrepearce) [inline artemis-jms-client/pom.xml:55]: Why is this needed it comes as a transitive anyhow

Comment (jbertram) [reply artemis-jms-client/pom.xml:55]: Yes, it comes in as transitive, but `org.apache.activemq.artemis.jms.client.ActiveMQDestination` uses `org.apache.activemq.artemis.api.core.QueueConfiguration` (from `artemins-commons`) explicitly so IMO the dependency should be explicit as well instead of leaving it as transitive.

Comment (michaelandrepearce) [reply artemis-jms-client/pom.xml:55]: Fair enough +1
